olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationSettings.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <boost/optional.hpp>
+#include <memory>
+#include <string>
+
+#include <olp/core/http/NetworkProxySettings.h>
+
+#include "AuthenticationApi.h"
+
+namespace olp {
+namespace http {
+class Network;
+}
+
+namespace thread {
+class TaskScheduler;
+}
+
+
+namespace authentication {
+static const std::string kHereAccountProductionTokenUrl =
+    "https://account.api.here.com/oauth2/token";
```

Comment:
Correct me if I'm wrong, but defining static const strings in headers might cause issues with linkers, unspecified global initialization errors, as well as, sometimes it causes even double-free segfaults. I suggest moving initialization in a cpp file.

Reply:
Agree.
Settings.h using the same url `https://account.api.here.com/oauth2/token`. Would be good to move both to one place. Do we have such place?
Or make token_endpoint_url `boost::optional` as you mention below. AuthenticationClient will use `https://account.api.here.com/oauth2/token` instead of `boost::none`.

Reply:
Use `static constexpr auto kHereAccountProductionTokenUrl = "...."`
This is a compile time define.
If this also does not work then let it empty and write comment that if empty we will use "https://account.api.here.com/oauth2/token" although this is not my favorite solution and is pretty ambiguous.